ViewTxt/ViewTxt/Form1.cs

```diff
@@ -0,0 +1,211 @@
+            Button1.Size = new Size(111, 31);
+            Button1.TabIndex = 3;
+            Button1.Text = "Выбрать каталог";
+            Button1.UseVisualStyleBackColor = true;
+            Button1.Click += (a, b) =>
+            {
+                if (FolderBrowserDialog1.ShowDialog() != DialogResult.OK)
+                    return;
+                Label1.Text = FolderBrowserDialog1.SelectedPath;
+                var directoryInfo = new DirectoryInfo(FolderBrowserDialog1.SelectedPath);
+                ListBox1.Items.Clear();
+                ListBox1.SelectedIndex = -1;
+                try
+                {
+                    foreach (FileInfo enumerateFile in directoryInfo.EnumerateFiles("*.txt"))
+                        ListBox1.Items.Add(enumerateFile.Name);
+                }
+                finally{}
+            };
+            Controls.Add(Button1);
+        }
+
+        private void CreateListBox()
+        {
+            ListBox1 = new ListBox();
+            ListBox1.FormattingEnabled = true;
+            ListBox1.Location = new Point(16, 510);
+            ListBox1.Size = new Size(380, 160);
+            ListBox1.Sorted = true;
+            ListBox1.TabIndex = 2;
+            ListBox1.SelectedIndexChanged += (a, b) =>
+            {
+                var path = Path.Combine(Label1.Text, ListBox1.SelectedItem.ToString());
+                var txt = "";
+                using (StreamReader str = new StreamReader(path))
+                {
+                    txt = str.ReadToEnd();
+                }
+                TextBox1.Text = CurrentEncoding.GetString(Encoding.UTF8.GetBytes(txt));
+            };
+            Controls.Add(ListBox1);
+        }
+
+        private void CreateLabel()
+        {
+            Label1 = new Label();
+            Label1.Location = new Point(18, 471);
+            Label1.Size = new Size(510, 22);
+            Label1.TabIndex = 1;
+            Controls.Add(Label1);
+        }
+
+        private void ChooseEncoding(object sender, EventArgs e)
+        {
+            string text = ((ButtonBase)sender).Text;
+            string Left = text;
+            Encoding encoding = Operators.CompareString(Left, "Зап.-европ.", false) != 0 ? (Operators.CompareString(Left, "Юникод", false) != 0 ? Encoding.GetEncoding(1251) : Encoding.GetEncoding("utf-8")) : Encoding.GetEncoding(1250);
+            if (Operators.CompareString(CurrentEncodingDesc, text, false) == 0)
+                return;
+            CurrentEncoding = encoding;
+            CurrentEncodingDesc = text;
+            if (Operators.CompareString(TextBox1.Text, "", false) != 0)
+            {
+                var path = Path.Combine(Label1.Text, ListBox1.SelectedItem.ToString());
+                var txt = "";
+                using (StreamReader str = new StreamReader(path))
+                {
+                    txt = str.ReadToEnd();
+                }
+                TextBox1.Text = CurrentEncoding.GetString(Encoding.UTF8.GetBytes(txt));
```

Comment:
В вашей редакции работает только юникод. Да и зачем так сложно. Можно сразу открыть поток на чтение в нужной кодировке
```suggestion
                using (StreamReader str = new StreamReader(path, CurrentEncoding))
                {
                    txt = str.ReadToEnd();
                }
                TextBox1.Text = txt;
```